sdmetrics/detection/tabular/base.py

```diff
@@ -35,12 +35,27 @@ def predict_proba(self, X):
         """
         raise NotImplementedError()
 
-    def metrics(self, metadata, real_tables, fake_tables):
+    def metrics(self, metadata, real_tables, synthetic_tables):
+        """
+        This function yields a sequence of Metric object.
+
+        Args:
+            metadata (sdv.Metadata): The Metadata object from SDV.
+            real_tables (dict): A dictionary mapping table names to dataframes.
+            synthetic_tables (dict): A dictionary mapping table names to dataframes.
+
+        Yields:
+            Metric: The next metric.
+        """
+        yield from self._single_table(metadata, real_tables, synthetic_tables)
+        yield from self._parent_child(metadata, real_tables, synthetic_tables)
+
+    def _single_table(self, metadata, real_tables, synthetic_tables):
```

Comment:
name it _single_table_detection perhaps?

Reply:
Done.